### PR TITLE
Rely on standard utils for `IsSubPath`

### PR DESCRIPTION
**Pull request by corneliusweig**
* Add more test cases
* Fix misleading argument naming of `IsSubPath`
       - subPath -> basePath
       - path    -> subPath

### CODE REVIEW

File: pkg/pathutil/pathutil_test.go [hunk 1]
```diff
@@ -55,17 +56,54 @@ func TestIsSubPathExtending(t *testing.T) {
...
 				extendingPath: filepath.Join("a"),
 			},
-			wantExtending:   nil,
-			wantIsExtending: false,
+		},
+		{
+			name: "base path is not clean",
+			args: args{
+				basePath:      filepath.Join("d", "..", "a", "b", "..", "..", "a"),
+				extendingPath: filepath.Join("a", "b"),
```
Comment (ahmetb) [inline pkg/pathutil/pathutil_test.go]: nit: test cases are hard to read
do you mind quickly replacing these with `filepath.FromSlash()`
maybe not needed so much I'm not sure

**Comment by ahmetb** [comment]
/lgtm
/approve
/hold
feel free to /hold cancel as you wish if you wanna fix nit above

File: pkg/pathutil/pathutil_test.go (hunk 1 above)
Comment (corneliusweig) [reply pkg/pathutil/pathutil_test.go]: Oh it's a good idea and easily done.

**Pull request merged**